src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React, { Component } from 'react'
+
+interface Props {
+  children: React.ReactNode
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean
+  errorInfo?: React.ErrorInfo
+}
+
+export class ErrorBoundary extends Component<Props, ErrorBoundaryState> {
+  state: ErrorBoundaryState = { hasError: false }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  static getDerivedStateFromError(error: Error) {
+    return { hasError: true }
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    this.setState({ errorInfo })
+  }
+
+  render() {
+    const { hasError, errorInfo } = this.state
+
+    if (hasError) {
+      // Render fallback UI
+      return (
+        <div
+          style={{
+            border: '1px solid crimson',
+            borderRadius: '7px',
+            padding: '12px',
+            color: 'crimson'
+          }}
+        >
+          <span>
+            Unexpected error occurred. You may want to reload the page and try it
+            again.&nbsp;
+            <span
+              style={{ cursor: 'pointer', color: '#0077FF' }}
+              onClick={() => {
+                window.location.reload()
+              }}
+            >
+              Reload
+            </span>
+          </span>
+          {errorInfo && (
+            <div style={{ marginLeft: '2px', marginTop: '7px' }}>
+              <details>
+                <summary>Click for error details (developers use only)</summary>
+                <code style={{ whiteSpace: 'pre' }}>{errorInfo.componentStack}</code>
```

Comment:
we might want to rethink this. It's not recommended to print stack traces in prod when errors occur, because it tends to leak application/architecture information that could be used against us by a malicious user. 
https://owasp.org/www-community/Improper_Error_Handling

Reply:
Yeah,I was kind of ignorant on this since our codebase is open to the public. I will remove it!